Compiler/BackEnd/Tearing.mo

```diff
@@ -2911,8 +2911,15 @@ algorithm
   if debug then execStat("Tearing.ModifiedCellierHeuristic_3 - 2"); end if;
 
   // 2. Determine the variables in causEq
-  msel := Array.select(mIn,causEq);
-  potentialTVars := List.unique(List.flatten(arrayList(msel)));
+  //msel := Array.select(mIn,causEq);
+
+  potentialTVars := {};
+  for e in causEq loop
+    potentialTVars := listAppend(arrayGet(mIn, e), potentialTVars);
+  end for;
+  potentialTVars := List.unique(potentialTVars);
```

Comment:
Sounds reasonable. But using AvlSetInt seems to eat up about half of the saved memory consumption.